src/wayland/output/xdg.rs

```diff
@@ -0,0 +1,177 @@
+
+        xdg_output.logical_position(inner.logical_position.x, inner.logical_position.y);
+
+        if let Some(size) = inner.physical_size {
+            let logical_size = size.to_logical(inner.scale);
+            xdg_output.logical_size(logical_size.w, logical_size.h);
+        }
+
+        if xdg_output.as_ref().version() >= 2 {
+            xdg_output.name(inner.name.clone());
+            xdg_output.description(inner.description.clone());
+        }
+
+        // xdg_output.done() is deprecated since version 3
+        if xdg_output.as_ref().version() < 3 {
+            xdg_output.done();
+        }
+
+        wl_output.done();
+
+        xdg_output.quick_assign(|_, _, _| {});
+        xdg_output.assign_destructor(Filter::new(|xdg_output: ZxdgOutputV1, _, _| {
+            let inner = &xdg_output.as_ref().user_data().get::<XdgOutput>().unwrap().inner;
+            inner
+                .lock()
+                .unwrap()
+                .instances
+                .retain(|o| !o.as_ref().equals(&xdg_output.as_ref()));
+        }));
+        xdg_output.as_ref().user_data().set_threadsafe({
+            let xdg_output = self.clone();
+            move || xdg_output
+        });
+
+        inner.instances.push(xdg_output.deref().clone());
+    }
+
+    pub(super) fn change_current_state(
+        &self,
+        new_mode: Option<Mode>,
+        new_scale: Option<i32>,
+        new_location: Option<Point<i32, Logical>>,
+    ) {
+        let mut output = self.inner.lock().unwrap();
+
+        if let Some(new_mode) = new_mode {
+            output.physical_size = Some(new_mode.size);
+        }
+        if let Some(new_scale) = new_scale {
+            output.scale = new_scale;
+        }
+        if let Some(new_location) = new_location {
+            output.logical_position = new_location;
+        }
+
+        for instance in output.instances.iter() {
+            if new_mode.is_some() | new_scale.is_some() {
+                if let Some(size) = output.physical_size {
+                    let logical_size = size.to_logical(output.scale);
+                    instance.logical_size(logical_size.w, logical_size.h);
+                }
+            }
+
+            if new_location.is_some() {
+                instance.logical_position(output.logical_position.x, output.logical_position.y);
+            }
+
+            // xdg_output.done() is deprecated since version 3
+            if instance.as_ref().version() < 3 {
+                instance.done();
+            }
+
+            // No need for wl_output.done() here, it will be called by caller (super::Output::change_current_state)
+        }
+    }
+}
+
+/// Initialize a xdg output manager global.
+pub fn init_xdg_output_manager<L>(display: &mut Display, logger: L) -> Global<ZxdgOutputManagerV1>
+where
+    L: Into<Option<::slog::Logger>>,
+{
+    let log = crate::slog_or_fallback(logger).new(o!("smithay_module" => "xdg_output_handler"));
+
+    display.create_global(
+        3,
```

Comment:
magic value? might be worth using the `VERSION` field defined in `ZxdgOutputManagerV1`

Reply:
Sorry, but I kinda don't follow. What do you mean by `VERSION field defined in ZxdgOutputManagerV1`? Do you mean version constants in [zxdg_output_manager_v1](https://docs.rs/wayland-protocols/0.28.6/wayland_protocols/unstable/xdg_output/v1/server/zxdg_output_manager_v1/index.html#constants) module? None of those are equal to the protocol version that I'm using.
(I probably just misunderstood or missed something)

Reply:
Yeah, it's not a magic value, it is the maximum version of the protocol currently supported by the code of this module, while the `VERSION` associated constant of `ZxdgOutputManagerV1` is the maximum version supported by the bindings generated by `wayland-server`. We should not use these `VERSION` constants directly in `create_global()` as this would automatically bump the version advertised to clients to the latest of the XML files, even if smithay has not been updated to support them yet.